src/market/market.module.ts

```diff
@@ -318,12 +370,16 @@ export class MarketModuleImpl implements MarketModule {
     );
   }
 
-  async negotiateProposal(receivedProposal: OfferProposal, offer: DemandSpecification): Promise<void> {
-    return this.deps.marketApi.counterProposal(receivedProposal, offer);
+  async negotiateProposal(offerProposal: OfferProposal, counterDemand: DemandSpecification): Promise<OfferProposal> {
+    const counterProposal = await this.deps.marketApi.counterProposal(offerProposal, counterDemand);
```

Comment:
I feel like counter proposal should be added to the repo in `negotiateProposal`, instead of the adapter doing that.